src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1088,7 +1088,7 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
 
             if (state.CancellationToken.IsCancellationRequested)
             {
-                state.Tcs.TrySetCanceled();
+                state.Tcs.TrySetCanceled(state.CancellationToken);
```

Comment:
Thanks for pointing that out. I'll investigate if the change is needed also there.